src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java

```diff
@@ -385,12 +386,29 @@
                     // Skip y-sub-flows.
                     continue;
                 }
-                deletionRequests.add(sendDeleteFlow(flow.getId(), requestId));
+                CompletableFuture<FlowResponsePayload> request = sendDeleteFlow(flow.getId(), requestId)
+                        .handle((value, possibleException) -> {
+                            if (possibleException != null) {
+                                failedFlows.add(flow.getId());
+                            }
+                            return value;
+                        });
+                deletionRequests.add(request);
             }
             return deletionRequests;
         }).thenApply(requests -> collectResponses(requests, FlowResponsePayload.class)
-                .thenApply(result::complete));
-
+                .thenApply(results -> {
+                    if (!failedFlows.isEmpty()) {
+                        StringBuilder errorBuilder = new StringBuilder("The following flows haven't been deleted: ");
+                        result.completeExceptionally(new MessageException(correlationId,
+                                System.currentTimeMillis(),
+                                ErrorType.REQUEST_INVALID,
+                                "Couldn't delete flow",
+                                errorBuilder.append(String.join(",", failedFlows))
```

Comment:
I think it's better to have `", "` instead of `","` but it's a minor comment

Reply:
Thanks, agree with you.